### PR TITLE
Minor fixes for Github workflows

### PR DESCRIPTION
**Pull request by csujedihy**
## Description

1. Skip installing signing certs if `ForTest` is not used with `ForKernel` in prepare-machine.ps1.
2. Only fetch the number of commits in checkout action required by the workflow.

## Testing

CI

### CODE REVIEW

File: scripts/prepare-machine.ps1 [hunk 1]
```diff
@@ -153,7 +153,13 @@ if ($ForTest) {
     # enabled for any possible test.
     $InstallTestCertificates = $true
     $InstallClog2Text = $true
-    $InstallSigningCertificates = $true; # For kernel drivers
+
+    # Since installing signing certs also checks whether test signing is enabled, which most
+    # likely will fail on a devbox, do it only when we need to test kernel drivers so that
+    # local testing setup won't be blocked by test signing not enabled.
+    if ($ForKernel) {
+        $InstallSigningCertificates = $true;
```
Comment (nibanks) [inline scripts/prepare-machine.ps1:161]: Don't we need for duonic too?

File: .github/workflows/build-reuse-darwin-framework.yml [hunk 2]
```diff
@@ -48,8 +48,6 @@ jobs:
     steps:
     - name: Checkout repository
       uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9
-      with:
-        fetch-depth: 0
```
Comment (nibanks) [inline .github/workflows/build-reuse-darwin-framework.yml:52]: I thought I had a reason for this...

Comment (csujedihy) [reply .github/workflows/build-reuse-darwin-framework.yml:52]: If a workflow needs to access tags, commit history or other branches than the ref branch, then it's needed.

File: scripts/prepare-machine.ps1 (hunk 1 above)
Comment (csujedihy) [reply scripts/prepare-machine.ps1:161]: We have a check for duonic later on.

```
if ($InstallDuoNic) {
    $InstallSigningCertificates = $true;
}
```

**Review by nibanks — APPROVED** [review]

**Pull request merged**